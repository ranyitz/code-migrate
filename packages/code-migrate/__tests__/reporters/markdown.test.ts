import path from 'path';
import { resolveFixture, runMigrationAndGetOutput } from '../utils';
import stripAnsi from 'strip-ansi';
// TODO: Replace this package with native call after node v.15
// @ts-expect-error no types
import replaceAll from 'string.prototype.replaceall';

const stripDynamicContent = (output: string) => {
  output = output.replace(/> \/([\w/\-_]+)/, '> /static-directory');

  output = replaceAll(
    output,
    / {3}at( [\w.<>\d~!_:-]+)? \(?[\w.<>\d~!_/\\:-]+\)?/gim,
    '   at function (/path/to/file)'
  );

  return output;
};

describe('markdown reporter', () => {
  it('passing', async () => {
    const fixtures = resolveFixture('transform');
    const migrationFile = path.join(fixtures, 'migration.ts');

    let output = stripAnsi(
      await runMigrationAndGetOutput({
        fixtures,
        migrationFile,
        reporterName: 'markdown',
      })
    );

    expect(stripDynamicContent(output)).toMatchSnapshot();
  });

  it('error', async () => {
    const fixtures = resolveFixture('error');
    const migrationFile = path.join(fixtures, 'migration.ts');

    const output = stripAnsi(
      await runMigrationAndGetOutput({
        fixtures,
        migrationFile,
        reporterName: 'markdown',
      })
    );

    expect(stripDynamicContent(output)).toMatchSnapshot();
  });
});
