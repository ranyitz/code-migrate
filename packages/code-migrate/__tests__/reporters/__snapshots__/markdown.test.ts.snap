// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`markdown reporter error 1`] = `
"
## error
> /static-directory

### ⚠️  The following migration tasks were failed

#### transform bar to baz in json contents
**ERROR** \`baz.json\`
\`\`\`
SyntaxError: Unexpected token } in JSON at position 18
    at function (/path/to/file)
    at function (/path/to/file)
    at function (/path/to/file)
    at function (/path/to/file)
    at function (/path/to/file)
    at function (/path/to/file)
    at function (/path/to/file)
    at function (/path/to/file)
    at function (/path/to/file)
    at function (/path/to/file)
\`\`\`

**🤷‍ No changes have been made**"
`;

exports[`markdown reporter passing 1`] = `
"
## transform
> /static-directory

#### transform bar to baz in json contents
\`\`\`
baz.json
\`\`\`"
`;
